lib/video_info/provider.rb

```diff
@@ -75,8 +75,13 @@ def _clean_options(options)
 
     def _set_data_from_api(api_url = _api_url)
       _set_data_from_api_impl(api_url)
-      rescue OpenURI::HTTPError, *NetHttpTimeoutErrors.all
-        raise VideoInfo::HttpError.new 'unexpected network error while fetching information about the video'
+    rescue OpenURI::HTTPError, *NetHttpTimeoutErrors.all => e
+      if e.instance_of?(OpenURI::HTTPError) && e.respond_to?(:io) && e.io.respond_to?(:status)
+        response_code = e.io.status[0]
+        VideoInfo.logger.warn('your API key is probably invalid. Please verify it.') if response_code == "400"
```

Comment:
Line is too long. [110/80]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.